File: src/Entity/Item.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Item
 *
 * @ORM\Table(name="item", indexes={@ORM\Index(name="author", columns={"author"})})
 * @ORM\Entity
 */
class Item
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="bigint", nullable=false, options={"unsigned"=true})
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=255, nullable=false)
     */
    private $name;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="creationDate", type="datetime", nullable=false)
     */
    private $creationdate;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="lastModified", type="datetime", nullable=true)
     */
    private $lastmodified;

    /**
     * @var int|null
     *
     * @ORM\Column(name="modifiedBy", type="bigint", nullable=true, options={"unsigned"=true})
     */
    private $modifiedby;

    /**
     * @var bool|null
     *
     * @ORM\Column(name="active", type="boolean", nullable=true)
     */
    private $active;

    /**
     * @var string|null
     *
     * @ORM\Column(name="image", type="string", length=255, nullable=true)
     */
    private $image;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="author", referencedColumnName="id")
     * })
     */
    private $author;

    /**
     * @var \Doctrine\Common\Collections\Collection
     *
     * @ORM\ManyToMany(targetEntity="Collection", mappedBy="itemid")
     */
    private $collectionid;

    /**
     * Constructor
     */
    public function __construct()
    {
        $this->collectionid = new \Doctrine\Common\Collections\ArrayCollection();
    }

    public function getId(){
		return $this->id;
	}

	public function setId($id){
		$this->id = $id;
	}

	public function getName(){
		return $this->name;
	}

	public function setName($name){
		$this->name = $name;
	}

	public function getCreationdate(){
		return $this->creationdate;
	}

	public function setCreationdate($creationdate){
		$this->creationdate = $creationdate;
	}

	public function getLastmodified(){
		return $this->lastmodified;
	}

	public function setLastmodified($lastmodified){
		$this->lastmodified = $lastmodified;
	}

	public function getModifiedby(){
		return $this->modifiedby;
	}

	public function setModifiedby($modifiedby){
		$this->modifiedby = $modifiedby;
	}

	public function getActive(){
		return $this->active;
	}

	public function setActive($active){
		$this->active = $active;
	}

	public function getImage(){
		return $this->image;
	}

	public function setImage($image){
		$this->image = $image;
	}

	public function getAuthor(){
		return $this->author;
	}

	public function setAuthor($author){
		$this->author = $author;
	}

	public function getCollectionid(){
		return $this->collectionid;
	}

	public function setCollectionid($collectionid){
		$this->collectionid = $collectionid;
	}

}

File: api/src/main/java/be/jimmyd/cm/entities/Item.java
package be.jimmyd.cm.entities;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "cm_item")
public class Item {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "image")
    private String image;

    @Basic
    @Column(name = "creationdate")
    private LocalDateTime creationDate;

    @Basic
    @Column(name = "lastmodified")
    private LocalDateTime lastModified;

    @Basic
    @Column(name = "modifiedby")
    private Long modifiedBy;

    @Basic
    @Column(name = "active")
    private Boolean active;

    @ManyToMany
    @JoinTable(
            name = "cm_collectionitem",
            joinColumns = {@JoinColumn(name = "itemid")},
            inverseJoinColumns = {@JoinColumn(name = "collectionid")}
    )
    private List<Collection> collections = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "author", referencedColumnName = "id", nullable = false)
    private User author;

    @OneToMany(cascade = CascadeType.REMOVE)
    @JoinColumn(name = "itemid")
    private List<Itemdata> itemdata;

    private Item(){

    }

    private Item(Builder builder) {
        id = builder.id;
        name = builder.name;
        image = builder.image;
        creationDate = builder.creationDate;
        lastModified = builder.lastModified;
        modifiedBy = builder.modifiedBy;
        active = builder.active;
        collections = builder.collections == null ? new ArrayList<>(): builder.collections;
        author = builder.author;
        itemdata = builder.itemdata;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImage() {
        return image;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public LocalDateTime getLastModified() {
        return lastModified;
    }

    public Long getModifiedBy() {
        return modifiedBy;
    }

    public Boolean getActive() {
        return active;
    }

    public List<Collection> getCollections() {
        return collections;
    }

    public User getAuthor() {
        return author;
    }

    public List<Itemdata> getItemdata() {
        return itemdata;
    }


    public static final class Builder {
        private long id;
        private String name;
        private String image;
        private LocalDateTime creationDate;
        private LocalDateTime lastModified;
        private Long modifiedBy;
        private Boolean active;
        private List<Collection> collections;
        private User author;
        private List<Itemdata> itemdata;

        public Builder() {
        }

        public Builder withId(long val) {
            id = val;
            return this;
        }

        public Builder withName(String val) {
            name = val;
            return this;
        }

        public Builder withImage(String val) {
            image = val;
            return this;
        }

        public Builder withCreationDate(LocalDateTime val) {
            creationDate = val;
            return this;
        }

        public Builder withLastModified(LocalDateTime val) {
            lastModified = val;
            return this;
        }

        public Builder withModifiedBy(Long val) {
            modifiedBy = val;
            return this;
        }

        public Builder withActive(Boolean val) {
            active = val;
            return this;
        }

        public Builder withCollections(List<Collection> val) {
            collections = val;
            return this;
        }

        public Builder withAuthor(User val) {
            author = val;
            return this;
        }

        public Builder withItemdata(List<Itemdata> val) {
            itemdata = val;
            return this;
        }

        public Item build() {
            return new Item(this);
        }
    }
}
